src/Integration/Telemetry/TelemetryHelper.cs

```diff
@@ -82,7 +83,8 @@ public static bool IsSonarCloud(Uri sonarqubeUri)
                 InstallDate = telemetryData.InstallationDate,
                 Analyses = telemetryData.Analyses,
                 ShowHotspot = telemetryData.ShowHotspot,
-                TaintVulnerabilities = telemetryData.TaintVulnerabilities
+                TaintVulnerabilities = telemetryData.TaintVulnerabilities,
+                ServerNotifications = isConnected ? telemetryData.ServerNotifications : null
```

Comment:
This is a bit hacky, but with the code being scattered all over the place, I didn't find a better place for it.